addons/sales_team/tests/test_sales_team.py

```diff
@@ -91,6 +87,14 @@ def test_default_team_fallback(self):
             team = self.env['crm.team']._get_default_team_id()
             self.assertEqual(team, self.team_c2)
 
+        # multi company: prefer active company
+        self.user_sales_leads.write({
+            'company_ids': [(4, self.company_main.id)]
+        })
```

Comment:
Instead I would just modify the default company of the user (`company_id`), and even make sure there are multiple companies the user could pick from.

And then check against `self.env.company` explicitly so we know that's what we're expecting